src/main/java/tutorease/address/logic/commands/DeleteContactCommand.java

```diff
@@ -52,12 +54,23 @@ public CommandResult execute(Model model) throws CommandException {
         if (targetIndex.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
-
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+
+        deleteLessonsForStudent(model, personToDelete);
         model.deletePerson(personToDelete);
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, format(personToDelete)));
     }
 
+    private void deleteLessonsForStudent(Model model, Person student) {
+        // Collect lessons associated with the student
+        List<Lesson> lessonsToDelete = model.getFilteredLessonList().stream()
```

Comment:
Should we find lessons to delete from the entire list? As we are trying to delete everything.